### DISCUSSION TITLE
Server instrumentation / OpenMetrics support PoC

### DISCUSSION THREAD

**Discussion by hoffie** [jamulussoftware/jamulus#2440] [Ideas]
*This is still early PoC stuff*

I've hacked up a Python-based OpenMetrics exporter for Jamulus which uses the new (not yet merged) JSON-RPC interface to gather statistics about server internals.

Its main use case is performance analysis, but it may be extended over time. I'm currently using it to measure the mixing duration in the sever OnTimer loop to observe performance changes with different threading options.

I'm *not* planning to submit this as a PR as-is. The Python exporter should likely be a standalone repo. The Jamulus-related changes should probably be hidden behind some `#ifdef INSTRUMENTATION`.

The current state is here:
https://github.com/hoffie/jamulus/compare/dtinth:json-rpc...hoffie:server-instrumentation

**Comment by pljones** [comment]
Not looked but I think the WorldJam folks were looking for some more details on server usage.
